harness/tests/tensorboard/test_s3.py

```diff
@@ -37,16 +59,56 @@ def test_s3_build_missing_param() -> None:
 
     with pytest.raises(KeyError):
         env = test_util.get_dummy_env()
-        tensorboard.build(env.det_cluster_id, env.det_experiment_id, env.det_trial_id, conf)
+        tensorboard.build(
+            env.det_cluster_id,
+            env.det_experiment_id,
+            env.det_trial_id,
+            conf,
+            async_upload=False,
+        )
+
+
+@pytest.mark.parametrize("prefix", [None, "my/test/prefix/"])
+def test_s3_lifecycle_sync(monkeypatch: monkeypatch.MonkeyPatch, prefix: Optional[str]) -> None:
```

Comment:
why is this a separate test, rather than just adding an async=True/False parameter to the existing `test_s3_lifecycle` test?

Reply:
Agreed!